test/azure/terraform_azure_keyvault_example_test.go

```diff
@@ -0,0 +1,56 @@
+// +build azure
+
+// NOTE: We use build tags to differentiate azure testing because we currently do not have azure access setup for
+// CircleCI.
+
+package test
+
+import (
+	"testing"
+
+	"github.com/gruntwork-io/terratest/modules/azure"
+	"github.com/gruntwork-io/terratest/modules/random"
+	"github.com/gruntwork-io/terratest/modules/terraform"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestTerraformAzureKeyVaultExample(t *testing.T) {
+	t.Parallel()
+
+	uniquePostfix := random.UniqueId()
+
+	//expectedSecretName := fmt.Sprintf("secret-%s", uniquePostfix)
+	//expectedKeyName := fmt.Sprintf("key-%s", uniquePostfix)
+	//expectedCertificateName := fmt.Sprintf("cert-%s", uniquePostfix)
+
+	// website::tag::1:: Configure Terraform setting up a path to Terraform code.
+	terraformOptions := &terraform.Options{
+		// The path to where our Terraform code is located
+		TerraformDir: "../../examples/azure/terraform-azure-keyvault-example",
+		Vars: map[string]interface{}{
+			"postfix": uniquePostfix,
+		},
+	}
+
+	// website::tag::4:: At the end of the test, run `terraform destroy` to clean up any resources that were created
+	defer terraform.Destroy(t, terraformOptions)
+
+	// website::tag::2:: Run `terraform init` and `terraform apply`. Fail the test if there are any errors.
+	terraform.InitAndApply(t, terraformOptions)
+
+	// website::tag::3:: Run `terraform output` to get the values of output variables
+	keyVaultName := terraform.Output(t, terraformOptions, "key_vault_name")
+	secretName := terraform.Output(t, terraformOptions, "secret_name")
+	keyName := terraform.Output(t, terraformOptions, "key_name")
+	certificateName := terraform.Output(t, terraformOptions, "certificate_name")
+
+	// website::tag::4:: Determine whether the secret, key, and certificate exists
+	secretExists := azure.KeyVaultSecretExists(t, keyVaultName, secretName)
```

Comment:
Asserts should be using expectedSecret/Key/Certificate names exist.  See https://github.com/gruntwork-io/terratest/blob/48e24df893c925adef331c342b7d3979da08f51c/test/azure/terraform_azure_availabilityset_example_test.go#L24 and https://github.com/gruntwork-io/terratest/blob/48e24df893c925adef331c342b7d3979da08f51c/test/azure/terraform_azure_availabilityset_example_test.go#L51 for example